### PR TITLE
fix: page navigation during QR accounts selection

### LINKED ISSUE #10300: [Bug] User unable to go to the previous page on QR Account list
Bitrise build:- https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/771cadcd-68a8-4abf-b5ec-365a83eb224f 
OS:- iPhone 13 / iOS 17.5.1

1. recover from SRP
2. Add hardware wallet
2. Add QR hardware wallet (Keystone)
3. Click 'Prev' button

Expected:- It does not do anything as user is on the first page
Actual:- It navigates to the next page. User is unable to go back to any page. 

[Private Zenhub Video](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMUhiQlE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--15f69b8a0292921d4cc9663cc24806745a956518/RPReplay_Final1721031006.MP4)

### PR DESCRIPTION
**Pull request by dawnseeker8**
## **Description**
There is a bug in `ConnectQRHardware/index.tsx` page, `prevPage` method has pass wrong parameter to `initiateQRHardwareConnection()` method to make the `previous page` button click go to `next page`.

This PR will fix above issue by passing the correct parameter to above method.

this PR contain following changes:
1. create `PAGINATION_OPERATION` enum to centrally control 0: GET_FIRST_PAGE, 1: GET_NEXT_PAGE, -1: GET_PREVIOUS_PAGE.
2. refactor `LedgerSelectAccount` component and `ConnectQRHardware` component to use above shared enum.
3. Implement some testId for `ConnectQRHardware` screen to make more accurate query of component. (getByText not working well in unit tests)
4. Implement unit tests for assert the component rendering correct address for `go first page`, `go next` and `go previous`.

## **Related issues**

Fixes: #10300 

## **Manual testing steps**

1. recover from SRP
2. Add hardware wallet
3. Add QR hardware wallet (Keystone)
4. Click 'Prev' button
5. User should see the previous page account lists.

### **Before**

[Private Zenhub Video](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMUhiQlE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--15f69b8a0292921d4cc9663cc24806745a956518/RPReplay_Final1721031006.MP4)

### **After**

https://github.com/user-attachments/assets/aa8a957d-3fc6-4da7-9ba3-8a8004f00bfb

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/Views/ConnectQRHardware/index.tsx [hunk 1]
```diff
@@ -274,7 +274,7 @@ const ConnectQRHardware = ({ navigation }: IConnectQRHardwareProps) => {
   const prevPage = useCallback(async () => {
     resetError();
     const [qrInteractions, connectQRHardwarePromise] =
-      await initiateQRHardwareConnection(1);
+      await initiateQRHardwareConnection(-1);
```
Comment (gantunesr) [inline app/components/Views/ConnectQRHardware/index.tsx:277]: @dawnseeker8 is there a way we can prevent this from happening by adding some unit tests?

**Review by gantunesr — COMMENTED** [review]
LGTM! Left a small question

File: app/components/Views/ConnectQRHardware/index.tsx (hunk 1 above)
Comment (dawnseeker8) [reply app/components/Views/ConnectQRHardware/index.tsx:277]: The original component doesn't have any unit tests file, however, i will implement a unit tests for this component to cover all `go previous`, `go next` and `go first`.

Comment (dawnseeker8) [reply app/components/Views/ConnectQRHardware/index.tsx]: hi, @gantunesr i have finished iimplementing the unit tests for `ConnectQRHardware` component, there are quite a lot of changes according to this change. 

1. create `PAGINATION_OPERATION` enum to centrally control 0: GET_FIRST_PAGE, 1: GET_NEXT_PAGE, -1: GET_PREVIOUS_PAGE.
2. refactor `LedgerSelectAccount` component and `ConnectQRHardware` component to use above shared enum.
3. Implement some testId for `ConnectQRHardware` screen to make more accurate query of component. (getByText not working well in unit tests)
4. Implement unit tests for assert the component rendering correct address for `go first page`, `go next` and `go previous`.

**Review by gantunesr — APPROVED** [review]

**Comment by vivek-consensys** [comment]
Tested using Bitrise build:- https://app.bitrise.io/app/be69d4368ee7e86d/build/93112a72-46f8-4228-bb12-b7c04eb39052/artifact/20cbd577b2557308/p/4ebb5cac35fb14f2dc3fdd021ce8f222
OS:- iPhone 13 / iOS 17.5.1

Working as expected. 

https://github.com/user-attachments/assets/a61fe3bc-44ef-4ab0-87e1-985a9c176892

**Pull request merged**